geometry_msgs/Pose2D CSP
std_msgs/UInt8 choice